test.js

```diff
@@ -11,36 +13,93 @@ var validate = require('./');
 
 describe('validate', function() {
   it('should validate input (lol, how meta)', function() {
-    try {
+    assert.throws(function _fixture() {
       validate('foo')
-    } catch(e) {
-      assert.equal(e.message, 'object-validate: schema should be an object')
-    }
-  });
+    }, /schema should be an object/)
+  })
 
   it('should return a function', function() {
     assert(typeof validate({}), 'function')
   })
 
   it('the returned function should validate input types', function() {
-    try {
-      validate({})('derp')
-    } catch(e) {
-      assert.equal(e.message, 'object-validate: data should be an object')
-    }
+    assert.throws(function _fixture() {
+      validate({})('foo')
+    }, /data should be an object/)
   })
 
   it('should validate objects', function() {
     var schema = validate({
-      name: function(val) { return typeof val == 'string' },
-      foo: function(val) { return typeof val == 'number' },
+      name: function(val) { return typeof val === 'string' },
+      age: function(val) { return typeof val === 'number' },
+    })
+
+    var res = schema({
+      name: 'foo',
+      age: 'bar'
+    })
+
+    assert.deepStrictEqual(res, { name: true, age: false })
+  })
+
+  it('should validate objects when validator class given', function() {
```

Comment:
> create a simple validator.

who says that it might be 100% simple? and why it would be?
And because these tests was missing it slow my PR and work.